### PR TITLE
[MicroGraph] Address ROCM warning and build failure

### PR DESCRIPTION
**Pull request by yuslepukhin**
### Description
Address build failures after Public API refactoring

### Motivation and Context
Make pipelines health.

### CODE REVIEW

File: onnxruntime/core/providers/migraphx/migraphx_execution_provider.cc [hunk 1]
```diff
@@ -1130,7 +1129,8 @@ Status MIGraphXExecutionProvider::Compile(const std::vector<FusedNodeAndGraph>&
             if (mgx_type != mgx_s.type()) {
               LOGS_DEFAULT(FATAL) << "MIGraphX: param type mismatch";
             }
-            m.add(name, migraphx::argument(param_shapes[name], input_tensor.GetTensorMutableRawData()));
+            m.add(name, migraphx::argument(param_shapes[name],
+                                           const_cast<void*>(input_tensor.GetTensorRawData())));
```
Comment (adrianlizarraga) [inline onnxruntime/core/providers/migraphx/migraphx_execution_provider.cc:1133]: It's a shame this const_cast can't be avoided (KernelContext::GetInput() must return a const/unowned input).

**Review by adrianlizarraga — APPROVED** [review]
API usage updates lgtm

File: onnxruntime/core/providers/migraphx/migraphx_execution_provider.cc (hunk 1 above)
Comment (yuslepukhin) [reply onnxruntime/core/providers/migraphx/migraphx_execution_provider.cc:1133]: It is not about our interfaces. It is about ROCM.

Comment (adrianlizarraga) [reply onnxruntime/core/providers/migraphx/migraphx_execution_provider.cc:1133]: Right. That's what I mean. Our interface is correctly returning const, but this function is not specifying const.

File: onnxruntime/core/providers/rocm/rocm_pch.h.orig [hunk 2]
```diff
@@ -0,0 +1,28 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
```
Comment (edgchen1) [inline onnxruntime/core/providers/rocm/rocm_pch.h.orig:1]: was this file unintentionally checked in?

Comment (yuslepukhin) [reply onnxruntime/core/providers/rocm/rocm_pch.h.orig]: Addressed

**Review by adrianlizarraga — APPROVED** [review]

**Review by edgchen1 — APPROVED** [review]

**Pull request merged**